src/LanguageServer/Impl/Program.cs

```diff
@@ -71,6 +76,24 @@ internal static class Program {
             }
         }
 
+        private static void EnableProfileOptimization() {
+            try {
+                // create directory for profile optimization
+                var path = Path.Combine(Path.GetTempPath(), "PythonLanguageServer", GetUserName(), "Profiles");
```

Comment:
I would be much happier if the caching directory code was moved to an IFileSystem method and used, rather than a tempdir or something custom.

Reply:
alright. I will make it to use cache folder than.

Reply:
For completeness, I mean to add something like `IFileSystem.GetCacheFolder`, move the code from `CacheFolderService` into `FileSystem` (sans logging?), then construct the FS earlier in Main to then use for profiling.

Reply:
hmmm.. cache folder require initialized to return cache folder path so I am not sure whether that is possible, but I can move it up to Initialize at least.

Reply:
Ah, I forgot about `cacheFolderPath`...